crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -341,12 +386,18 @@ def services
         else
           ::Crowbar::UpgradeStatus.new.end_step
         end
+      rescue StandardError => e
+        ::Crowbar::UpgradeStatus.new.end_step(
+          false,
+          services: {
+            data: e.message
```

Comment:
Could you add the usual `help: "Crowbar has failed. Check /var/log/crowbar/production.log for details."` here?